configtx/application_test.go

```diff
@@ -1874,10 +1940,58 @@ func TestSetApplicationOrgPoliciesFailures(t *testing.T) {
 		"TestPolicy": {},
 	}
 
-	err := c.Application().Organization("Org1").SetPolicies(AdminsPolicyKey, policies)
+	err := c.Application().Organization("Org1").SetPolicies(policies)
 	gt.Expect(err).To(MatchError("failed to set policies: unknown policy type: "))
 }
 
+func TestSetApplicationModPolicy(t *testing.T) {
+	t.Parallel()
+	gt := NewGomegaWithT(t)
+
+	channelGroup := newConfigGroup()
+	application, _ := baseApplication(t)
+
+	applicationGroup, err := newApplicationGroupTemplate(application)
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	channelGroup.Groups[ApplicationGroupKey] = applicationGroup
+	config := &cb.Config{
+		ChannelGroup: channelGroup,
+	}
+
+	c := New(config)
+
+	a := c.Application()
+	err = a.SetModPolicy("TestModPolicy")
+	gt.Expect(err).NotTo(HaveOccurred())
+
+	expectedPolicy := "TestModPolicy"
```

Comment:
nit: we can assert on the literal directly; no need for the temporary delcaration.